src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -413,17 +425,41 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         deviceId: string,
         encryptionKeyIndex: number,
         encryptionKeyString: string,
+        timestamp: number,
         delayBeforeUse = false,
     ): void {
         const keyBin = decodeBase64(encryptionKeyString);
 
         const participantId = getParticipantId(userId, deviceId);
-        const encryptionKeys = this.encryptionKeys.get(participantId) ?? [];
+        if (!this.encryptionKeys.has(participantId)) {
+            this.encryptionKeys.set(participantId, []);
+        }
```

Comment:
There must be a tidy way to avoid all of the timestamp / `keysEqual` checks for when it's known from the start that there's no existing key for the participant...but since that was the case even before this PR, we can revisit this later.